docs/modules/ROOT/pages/release-process.adoc

```diff
@@ -0,0 +1,155 @@
+= Release and Bundle Process
+
+This document describes the EC CLI release and bundle process for the Conforma project.
+
+== Overview
+
+The EC CLI release process is fully automated through GitHub Actions. It builds distribution binaries, container images, Tekton task bundles, and GitHub releases whenever changes are merged to the `main` branch.
```

Comment:
There's actually parallel GitHub and Konflux builds. Do you want to mention the Konflux build as well? There are arguments for and against I guess.